### PR TITLE
Add support for valuesFrom

### PR DESCRIPTION
**Pull request by StrongMonkey**
This PR adds support to define valueFrom in `fleet.yaml`, which allow user to put additional values.yaml into configmap/secrets. This is in case the valuesFile contains sensetive information and can't be exposed in git, user can prepopulate data into secrets in downstream clusters. 

https://github.com/rancher/fleet/issues/384

### CODE REVIEW

File: pkg/helmdeployer/deployer.go [hunk 1]
```diff
@@ -352,6 +356,102 @@ func (h *helm) install(bundleID string, manifest *manifest.Manifest, chart *char
...
+				return nil, err
+			}
+			data, ok := secret.Data[key]
+			if !ok {
+				return nil, fmt.Errorf("key %s is missing from secret %s/%s, can't use it in valuesFrom", key, namespace, name)
+			}
+			if err := yaml.Unmarshal(data, &val); err != nil {
+				return nil, err
+			}
+		} else if valuesFrom.ConfigMapKeyRef != nil {
```
Comment (nickgerace) [inline pkg/helmdeployer/deployer.go:396]: Don't we want to check both the Secret and the ConfigMap? Should this be a separate `if` block?

Comment (StrongMonkey) [reply pkg/helmdeployer/deployer.go:396]: The api is designed to only respect either secretRef or configmapRef. This followed the same valueFrom dedign from k8s pod api.

File: pkg/helmdeployer/deployer.go [hunk 2]
```diff
@@ -352,6 +356,102 @@ func (h *helm) install(bundleID string, manifest *manifest.Manifest, chart *char
...
+			name := valuesFrom.SecretKeyRef.Name
```
Comment (kinarashah) [inline pkg/helmdeployer/deployer.go:397]: We should read name and namespace form `valuesFrom.ConfigMapKeyRef`?

Comment (StrongMonkey) [reply pkg/helmdeployer/deployer.go:397]: That's a copy paste error :). Will fix

**Review by kinarashah — APPROVED** [review]

**Review by nickgerace — APPROVED** [review]

**Pull request merged**